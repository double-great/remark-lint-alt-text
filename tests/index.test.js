const remark = require("remark");
const dedent = require("dedent");
const plugin = require("../");

const processMarkdown = (markdown, opts) => {
  return remark().use(plugin, opts).process(markdown);
};

it("End in period", async () => {
  const lint = await processMarkdown(dedent`
      # Title of my site

      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

      ![alt texter](https://site.com/image.png)

      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    `);
  expect(lint.messages.length).toEqual(1);
  expect(lint.messages[0].reason).toMatchSnapshot();
});

it("Image is a link", async () => {
  const lint = await processMarkdown(dedent`
      [![Puppies.](https://site.com/image.png)](https://website.org)
    `);
  expect(lint.messages.length).toEqual(1);
  expect(lint.messages[0].reason).toMatchSnapshot();
});

it("Image is a link, should have alt text", async () => {
  const lint = await processMarkdown(dedent`
      [![](https://site.com/image.png)](https://website.org)
    `);
  expect(lint.messages.length).toEqual(1);
  expect(lint.messages[0].reason).toMatchSnapshot();
});

it("No warnings", async () => {
  const lint = await processMarkdown(dedent`
      # Title of my site


      ![A large black dog.](https://site.com/doggie.png)

      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

      ![A child holding a photograph.](https://site.com/kiddo.png)

    `);
  expect(lint.messages.length).toEqual(0);
});
