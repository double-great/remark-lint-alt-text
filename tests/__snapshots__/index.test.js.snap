// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End in period 1`] = `"Alt text should end with punctuation (https://git.io/JJk55)."`;

exports[`Image is a link 1`] = `"Images inside a link tag require alt text that describes the purpose of the link (https://git.io/JvfNj)."`;

exports[`Image is a link, should have alt text 1`] = `"Images inside a link tag require alt text that describes the purpose of the link (https://git.io/JvfNj)."`;
